### PR TITLE
✨ Eryone Ery32 mini (STM32F103VET6) board

### PR DESCRIPTION
**Pull request by augustozanellato**
### Description

This PR adds support for the Eryone Ery32 mini board used in the Eryone ER-20 printer.
This work is based on the files made available by Eryone [here](https://github.com/Eryone/STM32/tree/master/Marlin_ER20).
Please not that I never contributed to Marlin before so I'm not very knowledgeable on its coding standards or practices.

### Requirements

 * A compatible board (afaik it's only used in the ER-20 printer)

### Benefits

 * ER-20 users won't depend anymore on Eryone for firmware releases

### Configurations

[ER-20_bugfix-2.0.x_configuration.zip](https://github.com/MarlinFirmware/Marlin/files/7346192/ER-20_bugfix-2.0.x_configuration.zip)

### CODE REVIEW

**Comment by augustozanellato** [comment]
Should I also add a CI job for the board because it uses a custom platformio environment?

File: Marlin/src/pins/stm32f1/pins_ERYONE_ERY32_MINI.h [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    #define  E0_SLAVE_ADDRESS 0
+  #endif
+#endif
+//
+// Temperature Sensors
+//
+#define TEMP_BED_PIN       PC2  //TB
+#define TEMP_0_PIN         PC1  //TH1
+//#define TEMP_1_PIN         PC3  //TH2
+#define TEMP_CHAMBER_PIN   PC3
```
Comment (thisiskeithb) [inline Marlin/src/pins/stm32f1/pins_ERYONE_ERY32_MINI.h]: Is this the onboard/motherboard thermistor? If so, `TEMP_BOARD_PIN` would be used instead.

**Comment by ellensp** [comment]
Needs a non maple build env

**Comment by augustozanellato** [comment]
> Needs a non maple build env

I’m not sure why but with a non maple build env the Eryone flash tool would hang at 100% while flashing the board leaving it in a non working state

File: Marlin/src/pins/stm32f1/pins_ERYONE_ERY32_MINI.h (hunk 1 above)
Comment (augustozanellato) [reply Marlin/src/pins/stm32f1/pins_ERYONE_ERY32_MINI.h]: Yep it is, will update the PR in a couple hours

Comment (augustozanellato) [reply Marlin/src/pins/stm32f1/pins_ERYONE_ERY32_MINI.h]: PR updated to use TEMP_BOARD_PIN instead of TEMP_CHAMBER_PIN

**Comment by thisiskeithb** [comment]
> Needs a non maple build env

Even if it doesn't work, we don't want to lose it in the mix.

....just kidding. The PR was merged.

**Comment by augustozanellato** [comment]
> > Needs a non maple build env
> 
> Even if it doesn't work, we don't want to lose it in the mix.
@thisiskeithb if it’s needed I’ll make another pr for it

**Pull request merged**